src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -564,32 +548,56 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
                         if (_channel.State != CommunicationState.Opened)
                         {
                             // if we were aborted while getting our request or doing correlation, 
-                            // we need to abort the web request and bail
+                            // we need to abort the request and bail
                             Cleanup();
                             _channel.ThrowIfDisposedOrNotOpen();
                         }
 
-                        HttpOutput httpOutput = HttpOutput.CreateHttpOutput(_httpRequestMessage, _factory, request, _factory.IsChannelBindingSupportEnabled, _factory.GetHttpClient(), _factory._authenticationScheme, _timeoutHelper);
+                        bool suppressEntityBody = PrepareMessageHeaders(request);
 
+                        if (!suppressEntityBody)
+                        {
+                            _httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
+                        }
+
+                        await SendPreauthenticationHeadRequestIfNeeded();
                         bool success = false;
```

Comment:
If this is the case, would we be in a situation where the HEAD would be rejected, the POST would work, and we'd have to re-POST? Would that just be handled?